src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -74,6 +82,7 @@ class DRIVER_TYPE(str, Enum):
         "msodbcsql",
         "snowflake-snowpark-python",
     ],
+    DBMS_TYPE.MYSQL_DB: ["pymysql", "snowflake-snowpark-python"],
```

Comment:
let's add version cap, pymysql<=a,>=b
```suggestion
    DBMS_TYPE.MYSQL_DB: ["pymysql", "snowflake-snowpark-python"],
```